File: Leetcode/June Challenge/day 6/Queue Reconstruction by Height.java
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people,new Comparator<int[]>()
        { 
            public int compare(int []a,int []b)
            {
                if(a[0]==b[0])
                {
                    if(a[1]>b[1])
                        return 1;
                    return -1;
                }
                if(a[0]>b[0])
                    return 1;
                return -1;
            }
        });
        int [][]ans=new int[people.length][2];
        for(int i=0;i<people.length;i++)
        {
            // System.out.println(people[i][0]+" "+people[i][1]);
            ans[i][0]=-1;
            ans[i][1]=-1;
        }
        for(int i=0;i<people.length;i++)
        {
            int counter=people[i][1];
            for(int j=0;j<people.length;j++)
            {
                if(ans[j][0]==-1 && counter==0)
                {
                    ans[j][0]=people[i][0];
                    ans[j][1]=people[i][1];
                    break;
                }
                else if(ans[j][0]==-1 || ans[j][0]>=people[i][0])
                {
                    counter-=1;
                }
            }
        }
        return ans;
    }
}

File: Leetcode/June Challenge/day 6/Queue Reconstruction by Height.cpp
class Solution {
public:
    static bool c(vector<int>&a,vector<int>&b)
    {
        if(a[0]==b[0])
            return a[1]<b[1];
        return a[0]<b[0];
    }
    vector<vector<int>> reconstructQueue(vector<vector<int>>& p) {
        sort(p.begin(),p.end(),c);
        int cu;
        vector<vector<int>>ans(p.size(),vector<int>(2,-1));
        for(int i=0;i<p.size();i++)
        {
            cu=p[i][1];
            for(int j=0;j<p.size();j++)
            {
                if(cu==0 && ans[j][0]==-1)
                {
                    ans[j][0]=p[i][0];
                    ans[j][1]=p[i][1];
                    break;
                }
                else if(ans[j][0]==-1||ans[j][0]>=p[i][0])
                {
                    cu--;
                }
            }
        }
        return ans;
    }
};